drivers/eks/eks_driver.go

```diff
@@ -1279,7 +1297,21 @@ func (d *Driver) updateClusterAndWait(ctx context.Context, state state) error {
 			input.Name = aws.String(state.ClusterName)
 			output, err = svc.UpdateClusterVersionWithContext(ctx, input)
 		}
-
+		if invalidParam(err) {
+			describeInput := &eks.DescribeClusterInput{
+				Name: input.Name,
+			}
+			eksState, describeErr := svc.DescribeCluster(describeInput)
+			if eksState == nil || describeErr != nil {
+				// show original error encountered during update
+				return err
+			}
+			if eksState.Cluster.Version != nil && *eksState.Cluster.Version == state.KubernetesVersion {
```

Comment:
Cluster is a pointer also. Possible that eksState gets returned with nil cluster.